product_print_category/models/product_print_category_mixin.py

```diff
@@ -0,0 +1,23 @@
+# Copyright (C) 2022-Today: La Louve (<http://www.lalouve.net/>)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+
+
+class ProductPrintCategoryMixin(models.AbstractModel):
+    _name = "product.print.category.mixin"
+
+    @api.multi
+    def _update_to_print_values(self, vals):
+        # This function work for item that are product.product and product.template
+        to_update_item_ids = []
+        # Set 'To print' if we change one field choosen in print_category
+        for item in self.filtered(lambda x: x.print_category_id):
+            triggering_fields = item.print_category_id.field_ids.mapped("name") + [
+                "print_category_id"
+            ]
+            if len(list(set(vals.keys()) & set(triggering_fields))):
+                to_update_item_ids.append(item.id)
+        to_update_items = self.browse(to_update_item_ids)
+        super(ProductPrintCategoryMixin, to_update_items).write({"to_print": True})
```

Comment:
Why do you write on `super(ProductPrintCategoryMixin, to_update_items)` rather than directly on `to_update_items`?

There is no write here so it won't change the behavior but I already had confusing bugs because of this kind of pattern.

Reply:
Ok I get it : it's called from the `write` of product.product. A comment about this would be nice.

Reply:
something like ``# Prevent an infinite loop by calling the function super`` ?